### ISSUE TITLE
hash sum mismatch for grpcio when installing firebase-admin

### ISSUE THREAD

**Issue by ritchiedc** [bennuttall/piwheels#174]
Package name: grpcio
Issue type: Broken package
Link to PyPI page: https://pypi.org/project/grpcio
Link to piwheels page: https://www.piwheels.org/project/grpcio
Version: 1.21.1
Python version: 3.7
More information: I am getting an error similar to issue #146/170. I am trying to install the firebase-admin package on a Raspberry Pi 3B using the new Buster distribution. I was able to install grpcio using the method described in issue #170 except I had to use the full name and not the wildcard in the pip3 command. I got the following error:

```
sudo pip3 install firebase-admin
.
.
.
THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE. If you have updated the package versions, please update the hashes. Otherwise, examine the package contents carefully; someone may have tampered with them.
    grpcio<2.0dev,>=1.8.2; extra == "grpc" from https://www.piwheels.org/simple/grpcio/grpcio-1.21.1-cp37-cp37m-linux_armv7l.whl#sha256=676495c9942aa7fcd69f9226d062fe124e232efd815e052d705c7b37575c853f (from google-api-core[grpc]<2.0.0dev,>= 1.7.0; platform_python_implementation != "PyPy"->firebase-admin):
        Expected sha256 676495c9942aa7fcd69f9226d062fe124e232efd815e052d705c7b37575c853f
             Got        9e7d34ea07ad668edaccf04945d2e558fc9de3ee0f5b888a07511dbafbee3ac6
```

**Comment by bennuttall** [comment]
As you've noticed we've had a few reports of this happening, but I can't understand why. When I wget that file and check its hash I get a match:

```
ben@gunter:~ $ vim grpcio-1.21.1-cp37-cp37m-linux_armv7l.whl^C
ben@gunter:~ $ sha256sum grpcio-1.21.1-cp37-cp37m-linux_armv7l.whl
676495c9942aa7fcd69f9226d062fe124e232efd815e052d705c7b37575c853f  grpcio-1.21.1-cp37-cp37m-linux_armv7l.whl
```

Maybe if you clear your pip cache, if that's where it's getting its hashes from?

Try `pip3 install grpcio --no-cache-dir`

**Comment by ritchiedc** [comment]
I did some more testing and found that the downloads were terminating before complete. But rather than saying it couldn't download it gave the hash mismatch error message. I do not have good quality internet service which may be part of the problem but at the same time grpcio was the only package (apt-get or pip3) that behaved this way. I connected my router thru my phone hotspot (I can set it to use WAN over WiFi) and the installation of grpcio was successful.
